kedro/templates/project/hooks/utils.py

```diff
@@ -1,133 +1,137 @@
-    "nbsphinx==0.8.1",
-    "sphinx-autodoc-typehints==1.11.1",
-    "sphinx_copybutton==0.3.1",
-    "ipykernel>=5.3, <7.0",
-    "Jinja2<3.1.0",
-    "myst-parser~=0.17.2",
-]
-"""
+test_pyproject_requirements = ["tool.pytest.ini_options", "tool.coverage.report"]
 
+docs_pyproject_requirements = ["project.optional-dependencies"]
 
-def setup_template_add_ons(selected_add_ons_list, requirements_file_path, pyproject_file_path, python_package_name):
-    """Removes directories and files related to unwanted addons from
-    a Kedro project template. Adds the necessary requirements for
-    the addons that were selected.
 
-    Args:
-        selected_add_ons_list: a list containing numbers from 1 to 5,
-            representing specific add-ons.
-        requirements_file_path: the path to the requirements.txt file.
-        pyproject_file_path: the path to the pyproject.toml file
-            located on the the root of the template.
-    """
-    if "Linting" not in selected_add_ons_list:
-        pass
+# Helper Functions
+def remove_from_file(file_path, content_to_remove):
+    with open(file_path, 'r') as file:
+        lines = file.readlines()
+
+    # Split the content to remove into lines and remove trailing whitespaces/newlines
+    content_to_remove_lines = [line.strip() for line in content_to_remove.split('\n')]
+
+    # Keep lines that are not in content_to_remove
+    lines = [line for line in lines if line.strip() not in content_to_remove_lines]
+
+    with open(file_path, 'w') as file:
+        file.writelines(lines)
+
+
+def remove_nested_section(data, nested_key):
+    keys = nested_key.split('.')
+    current_data = data
+    # Look for Parent section
+    for key in keys[:-1]:  # Iterate over all but last element
+        if key in current_data:
+            current_data = current_data[key]
+        else:
+            return  # Parent section not found, nothing to remove
+
+    # Remove the nested section and any empty parent sections
+    current_data.pop(keys[-1], None)  # Remove last element otherwise return None
+    for key in reversed(keys[:-1]):
+        parent_section = data
+        for key_part in keys[:keys.index(key)]:
+            parent_section = parent_section[key_part]
+        if not current_data:  # If the section is empty, remove it
+            parent_section.pop(key, None)
+            current_data = parent_section
+        else:
+            break  # If the section is not empty, stop removing
+
+
+def remove_from_toml(file_path, sections_to_remove):
+    # Load the TOML file
+    with open(file_path, 'r') as file:
+        data = toml.load(file)
+
+    # Remove the specified sections
+    for section in sections_to_remove:
+        remove_nested_section(data, section)
+
+    with open(file_path, 'w') as file:
+        toml.dump(data, file)
+
+
+def remove_dir(path):
+    if path.exists():
+        shutil.rmtree(str(path))
+
+
+def remove_file(path):
+    if path.exists():
+        path.unlink()
```

Comment:
minor question should these all be private methods? I think only the `selected_template_add_ons` should be public, but this is minor and we can address this separately in  https://github.com/kedro-org/kedro/issues/3264

Reply:
Yes they should be, only `setup_template_add_ons` and `sort_requirements` should be public.

Reply:
Made them private.